### ISSUE TITLE
Unhandled panic in DecisionWorker 

### ISSUE THREAD

**Issue by levicui** [uber-go/cadence-client#526]
cadence-client version: Commit ID: 6f75c62b74d531e4cf4d7c93d9af6403cafe598c

> 
> ...
>     "stack": "code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*baseWorker).processTask.func1\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_worker_base.go:276\nruntime.call32\n\t/usr/lib/go-1.10/src/runtime/asm_amd64.s:573\nruntime.gopanic\n\t/usr/lib/go-1.10/src/runtime/panic.go:502\nruntime.panicmem\n\t/usr/lib/go-1.10/src/runtime/panic.go:63\nruntime.sigpanic\n\t/usr/lib/go-1.10/src/runtime/signal_unix.go:388\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowTaskHandlerImpl).completeWorkflow\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_handlers.go:1069\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowExecutionContextImpl).CompleteDecisionTask\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_handlers.go:760\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowExecutionContextImpl).ProcessLocalActivityResult\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_handlers.go:740\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowTaskPoller).processLocalActivityResult\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_pollers.go:349\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowTaskPoller).ProcessTask\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_pollers.go:209\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*baseWorker).processTask\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_worker_base.go:285",
>     "WorkerType": "DecisionWorker",
> ...
> 
>     "caller": "internal/internal_worker_base.go:276",
>     "@timestamp": "2018-07-21T10:00:25.253Z",
>     "PanicError": "runtime error: invalid memory address or nil pointer dereference",
>     "PanicStack": "base worker for DecisionWorker [panic]:\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowTaskHandlerImpl).completeWorkflow(0xc424b2d500, 0xc49ae77280, 0x0, 0xc4fc0eb6b0, 0xc446acae20, 0x2, 0x2, 0x0, 0x0, 0xc420065790)\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_handlers.go:1069 +0x3a\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowExecutionContextImpl).CompleteDecisionTask(0xc4fc0eb6b0, 0xc446acae01, 0x2, 0x2)\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_handlers.go:760 +0x1f9\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowExecutionContextImpl).ProcessLocalActivityResult(0xc4fc0eb6b0, 0xc5652d6c90, 0xc4fc0eb6b0, 0x0, 0x0, 0xc57287e400)\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_handlers.go:740 +0x142\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowTaskPoller).processLocalActivityResult(0xc432d060a0, 0xc5652d6c90, 0x0, 0x0)\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_pollers.go:349 +0xca\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*workflowTaskPoller).ProcessTask(0xc432d060a0, 0x281e7a0, 0xc5652d6c90, 0xc57287e400, 0x2)\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_task_pollers.go:209 +0x5e\ncode.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*baseWorker).processTask(0xc4328e35f0, 0x281e7a0, 0xc5652d6c90)\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_worker_base.go:285 +0xd7\ncreated by code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal.(*baseWorker).runTaskDispatcher\n\t/home/udocker/goautobots-v2/.tmp/.goroot/src/code.uber.internal/odp/goautobots/vendor/go.uber.org/cadence/internal/internal_worker_base.go:236 +0xd9",
> ...

I can provide more details if needed.

**Comment by yiminc** [comment]
this is fixed.

**Comment by sagikazarmark** [comment]
I can still reproduce this issue:

```
time=2018-09-06T19:09:39.710+0200 level=error caller=internal/internal_worker_base.go:284 msg="Unhandled panic." Domain=banzaicloud2 TaskList=pipeline WorkerID=3246@Marks-MBP@pipeline WorkerType=DecisionWorker PanicError="runtime error: invalid memory address or nil pointer dereference" PanicStack="
base worker for DecisionWorker [panic]:
VENDOR/go.uber.org/cadence/internal.(*workflowTaskHandlerImpl).getOrCreateWorkflowContext.func1(0xc4210a7c10, 0xc4210a7c08, 0xc4204ce150, 0x40ba860, 0xc42031ba40)
VENDOR/go.uber.org/cadence/internal/internal_task_handlers.go:520 +0x76
VENDOR/go.uber.org/cadence/internal.(*workflowTaskHandlerImpl).getOrCreateWorkflowContext(0xc4204ce150, 0xc420c21e00, 0x4094320, 0xc420c19e00, 0xc4213d2630, 0x0, 0x0)
VENDOR/go.uber.org/cadence/internal/internal_task_handlers.go:567 +0x267
VENDOR/go.uber.org/cadence/internal.(*workflowTaskHandlerImpl).ProcessWorkflowTask(0xc4204ce150, 0xc420c21e00, 0x4094320, 0xc420c19e00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
VENDOR/go.uber.org/cadence/internal/internal_task_handlers.go:621 +0x19f
VENDOR/go.uber.org/cadence/internal.(*workflowTaskPoller).processWorkflowTask(0xc4202fa960, 0xc4215ff640, 0x0, 0x0)
VENDOR/go.uber.org/cadence/internal/internal_task_pollers.go:232 +0xaf
VENDOR/go.uber.org/cadence/internal.(*workflowTaskPoller).ProcessTask(0xc4202fa960, 0x3442200, 0xc4215ff640, 0x1, 0x0)
VENDOR/go.uber.org/cadence/internal/internal_task_pollers.go:212 +0xa8
VENDOR/go.uber.org/cadence/internal.(*baseWorker).processTask(0xc420b2cdd0, 0x3442200, 0xc4215ff640)
VENDOR/go.uber.org/cadence/internal/internal_worker_base.go:293 +0xd7
created by VENDOR/go.uber.org/cadence/internal.(*baseWorker).runTaskDispatcher
VENDOR/go.uber.org/cadence/internal/internal_worker_base.go:244 +0xd9"

```

**Comment by yiminc** [comment]
@sagikazarmark , could you help to see if the panic is fixed with the #565

**Comment by sagikazarmark** [comment]
@yiminc Tested it all day, can't see the panic anymore.
